polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -1490,15 +1509,18 @@ class RaiseSetNDRange
 
     // We will introduce the operation after the last annotation
     OpBuilder::InsertionGuard ig(rewriter);
-    LLVM::VarAnnotation lastAnnotation = arguments.back().getAnnotation();
+    LLVM::VarAnnotation lastAnnotation =
+        arguments.back().getAnnotations().back();
     rewriter.setInsertionPointAfter(lastAnnotation);
```

Comment:
Wondering if we need to do something smarter here, e.g. inserting into the common postdom of the annotations' blocks.

Reply:
Does the `sycl.host.schedule_kernel` pattern care whether the `set_nd_range` is in a given block?

Reply:
No.